File: 05Types/alias.rs
// `NanoSecond` 是 `u64` 的新名字。
type NanoSecond = u64;
type Inch = u64;

// 通过这个属性屏蔽警告。
#[allow(non_camel_case_types)]
type u64_t = u64;
// 试一试 ^ 移除上面那个属性

fn main() {
    // `NanoSecond` = `Inch` = `u64_t` = `u64`.
    let nanoseconds: NanoSecond = 5 as u64_t;
    let inches: Inch = 2 as u64_t;

    // 注意类型别名*并不能*提供额外的类型安全，因为别名*并不是*新的类型。
    println!("{} nanoseconds + {} inches = {} unit?",
             nanoseconds,
             inches,
             nanoseconds + inches);
}

//可以用 type 语句给已有的类型取个新的名字。类型的名字必须遵循驼峰命名法
//别名的主要用途是避免写出冗长的模板化代码（boilerplate code）。

File: 05Types/alias.cs
using System;

//注意以下的功能要在C# 12 版本的的新功能：类型的别名。
// `NanoSecond` 是 `u64` 的新名字。
using NanoSecond = ulong;
using Inch = ulong;
using u64_t = ulong;
// 试一试 ^ 移除上面那个属性
public class Program{
    public static void Main(string[] args)
    {
        Console.WriteLine("This is C# Language");
        // `NanoSecond` = `Inch` = `u64_t` = `u64`.
        NanoSecond  nanoseconds = 5 ;
        Inch        inches = 2 ;

        // 注意类型别名*并不能*提供额外的类型安全，因为别名*并不是*新的类型。
        Console.WriteLine("{0} nanoseconds + {1} inches = {2} unit?",
                nanoseconds,
                inches,
                nanoseconds + inches);
    }
}